components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java

```diff
@@ -246,8 +263,15 @@ private void retrieveAndDeployArtifacts(String tenantDomain) {
                 } catch (ArtifactSynchronizerException e) {
                     log.error("Error in Deploying APIs to gateway", e);
                 }
+                // Logic becomes too complex and less readable if we consolidated, better to keep the blocks separate, especially since it enhances code clarity and maintainability
```

Comment:
see if this line expands more than maximum column size

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12201/commits/1136a795360198388c6b4bd97029c20b0f898832

Thanks and Regards